structured-event-log/src/main/java/org/apache/pulsar/structuredeventlog/slf4j/Slf4jEvent.java

```diff
@@ -0,0 +1,206 @@
+        return this;
+    }
+
+    @Override
+    public Event resource(String key, Supplier<String> value) {
+        resources.resource(key, value);
+        return this;
+    }
+
+    @Override
+    public Event attr(String key, Object value) {
+        getAttributes().add(key);
+        getAttributes().add(value);
+        return this;
+    }
+
+    @Override
+    public Event attr(String key, Supplier<String> value) {
+        this.attr(key, (Object)value);
+        return this;
+    }
+
+    @Override
+    public Event exception(Throwable t) {
+        this.throwable = t;
+        return this;
+    }
+
+    @Override
+    public Event atError() {
+        this.level = Level.ERROR;
+        return this;
+    }
+
+    @Override
+    public Event atInfo() {
+        this.level = Level.INFO;
+        return this;
+    }
+
+    @Override
+    public Event atWarn() {
+        this.level = Level.WARN;
+        return this;
+    }
+
+    @Override
+    public void log(Enum<?> event) {
+        throw new UnsupportedOperationException("TODO");
+    }
+
+    @Override
+    public void log(String event) {
+        try {
+            MDC.put("id", id);
+            if (traceId != null) {
+                MDC.put("traceId", traceId);
+            }
+            if (parentId != null) {
+                MDC.put("parentId", parentId);
+            }
+            resources.forEach(MDC::put);
+            if (attributes != null) {
+                EventResourcesImpl.forEach(attributes, MDC::put);
+            }
+            Logger logger = LoggerFactory.getLogger("stevlog");
+            switch (level) {
+            case ERROR:
+                if (throwable != null) {
+                    logger.error(event, throwable);
+                } else {
+                    logger.error(event);
+                }
+                break;
+            case WARN:
+                if (throwable != null) {
+                    logger.warn(event, throwable);
+                } else {
+                    logger.warn(event);
+                }
+                break;
+            case INFO:
+            default:
+                if (throwable != null) {
+                    logger.info(event, throwable);
+                } else {
+                    logger.info(event);
+                }
+                break;
+            }
+        } finally {
+            MDC.clear();
+        }
+    }
+
+    @Override
+    public void stash() {
+        throw new UnsupportedOperationException("TODO");
+    }
+
+    private List<Object> getAttributes() {
+        if (attributes == null) {
+            attributes = new ArrayList<>();
+        }
+        return attributes;
+    }
+
+    static String randomId() {
+        return Long.toString(
+                ThreadLocalRandom.current().nextLong(0x100000000000000L,
+                                                     0xFFFFFFFFFFFFFFFL),
+                16);
+    }
+
+    enum Level {
+        INFO,
```

Comment:
We can also give the option to do structured debug logs, if someone choses to

Reply:
created a JIRA